packages/cli-repl/src/completer.spec.ts

```diff
@@ -0,0 +1,125 @@
+    it('matches a collection command', () => {
+      const i = 'db.shipwrecks.findAnd';
+      expect(write(i)).to.deep.equal([['db.shipwrecks.findAndModify'], i]);
+    });
+
+    it('returns all suggestions', () => {
+      const i = 'db.shipwrecks.';
+      const collComplete = Object.keys(shellTypes.Collection.attributes)
+      const adjusted = collComplete.map(c => `${i}${c}`)
+
+      expect(write(i)).to.deep.equal([adjusted, i]);
+    });
+
+    it('matches several collection commands', () => {
+      const i = 'db.shipwrecks.find';
+      expect(write(i)).to.deep.equal([
+        [
+          'db.shipwrecks.find', 'db.shipwrecks.findAndModify',
+          'db.shipwrecks.findOne', 'db.shipwrecks.findOneAndDelete',
+          'db.shipwrecks.findOneAndReplace', 'db.shipwrecks.findOneAndUpdate'
+        ], i]);
+    });
+
+    it('does not have a match', () => {
+      const i = 'db.shipwrecks.pr';
+      expect(write(i)).to.deep.equal([[], i]);
+    });
+  });
+
+  context('when context is collections and aggregation cursor', () => {
+    it('matches an aggregation cursor command', () => {
+      const i = 'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).has';
+      expect(write(i)).to.deep.equal([
+        ['db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).hasNext'], i]);
+    });
+
+    it('returns all suggestions', () => {
+      const i = 'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).';
+      const aggCursorComplete = Object.keys(shellTypes.AggregationCursor.attributes)
+      const adjusted = aggCursorComplete.map(c => `${i}${c}`)
+
+      expect(write(i)).to.deep.equal([adjusted, i]);
+    });
+
+    it('does not have a match', () => {
+      const i = 'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).w';
+      expect(write(i)).to.deep.equal([[], i]);
+    });
+
+    it('has several matches', () => {
+      const i = 'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).i';
+      expect(write(i)).to.deep.equal([
+        [
+          'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).isClosed',
+          'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).isExhausted',
+          'db.shipwrecks.aggregate([{$sort: {feature_type: 1}}]).itcount'
+        ], i]);
+    });
+  });
+
+  context('when context is collections and collection cursor', () => {
+    it('matches a collection cursor command', () => {
+      const i = 'db.shipwrecks.find({feature_type: "Wrecks - Visible"}).for';
+      expect(write(i)).to.deep.equal([
+        ['db.shipwrecks.find({feature_type: "Wrecks - Visible"}).forEach'], i]);
+    });
+
+    it('returns all suggestions', () => {
+      const i = 'db.shipwrecks.find({feature_type: "Wrecks - Visible"}).';
+      const collCursorComplete = Object.keys(shellTypes.Cursor.attributes);
+      const adjusted = collCursorComplete.map(c => `${i}${c}`)
+
+      expect(write(i)).to.deep.equal([adjusted, i]);
+    });
+
+    it('does not have a match', () => {
```

Comment:
I struggled a bit to understand the test case is it more 'returns empty suggestions when it does not have a match'? Maybe i still misunderstood it :).

Reply:
it's more of "does not return completions if there is no match" can rename it to that <- if it makes more sense.